llvm/lib/TargetParser/Triple.cpp

```diff
@@ -740,6 +742,19 @@ static Triple::SubArchType parseSubArch(StringRef SubArchName) {
         .EndsWith("v1.6", Triple::SPIRVSubArch_v16)
         .Default(Triple::NoSubArch);
 
+  if (SubArchName.starts_with("dxil"))
+    return StringSwitch<Triple::SubArchType>(SubArchName)
+        .EndsWith("v1.0", Triple::DXILSubArch_v10)
+        .EndsWith("v1.1", Triple::DXILSubArch_v11)
+        .EndsWith("v1.2", Triple::DXILSubArch_v12)
```

Comment:
does it matter than this would match something like "dxilborkborkborkv1.2"?

Reply:
> does it matter than this would match something like "dxilborkborkborkv1.2"?

Pushed a change to address this. Thanks!